sklearn/model_selection/_validation.py

```diff
@@ -256,19 +268,52 @@ def cross_validate(estimator, X, y=None, *, groups=None, scoring=None, cv=None,
     if return_estimator:
         ret['estimator'] = results["estimator"]
 
-    test_scores = _aggregate_score_dicts(results["test_scores"])
+    test_scores_dict = _normalize_score_results(results["test_scores"])
     if return_train_score:
-        train_scores = _aggregate_score_dicts(results["train_scores"])
+        train_scores_dict = _normalize_score_results(results["train_scores"])
 
-    for name in test_scores:
-        ret['test_%s' % name] = test_scores[name]
+    for name in test_scores_dict:
+        ret['test_%s' % name] = test_scores_dict[name]
         if return_train_score:
             key = 'train_%s' % name
-            ret[key] = train_scores[name]
+            ret[key] = train_scores_dict[name]
 
     return ret
 
 
+def _insert_error_scores(results, error_score):
+    """Insert error in results by replacing them with `error_score`.
```

Comment:
Could be worth mentioning that the insert is in place (even if it is almost obvious)